Adds the spontaneous brain trauma event

coiax opened the pull request:
:cl: coiax
add: Adds the Spontaneous Brain Trauma to the event pool. Sometimes your brain just goes a little wrong.
/:cl:

Brain traumas are !fun!, and even more !fun! than your appendix getting inflamed.

coiax committed:
Added spontaneous brain trauma event

coiax commented:
@XDTM

XDTM commented:
👍

improvedname commented:
Spontaneous chestburst event when

GuyonBroadway commented:
Randomly getting full body paralysis is a large amount of negative fun.

XDTM commented:
that's a really small chance though now that paralysis has 9 different outcomes only one of which is full body

Shadowflame909 commented:
Spontaneous Smiting event when.

Sometimes the gods just want to hit you with a bolt of lightning.

GuyonBroadway commented:
It's still a nonzero chance to have all control taken away from the player with 0 warning or means of countering it.


While it's really low there is still going to be a case where someone just drops paralysed and rightfully ahelps because "I'm stuck and can't do anything"

Eskjjlj commented:
Spontaneous nuke explosion when

XDTM commented on code/modules/events/brain_trauma.dm:
```diff
@@ -0,0 +1,34 @@
+		TRAUMA_RESILIENCE_BASIC = 50,
+		TRAUMA_RESILIENCE_SURGERY = 30,
+		TRAUMA_RESILIENCE_LOBOTOMY = 15,
+		TRAUAM_RESILIENCE_MAGIC = 5))
```
I think it's probably better to limit it to lobotomy, just so there's always something you can do about it

coiax replied:
Doesn't cloning remove all brain traumas anyway

k-k-karma replied:
it's funnier if you get magic, but trauma is spelt wrong on it?

XDTM replied:
oh yeah that's TRAUAM how is it passing the travis check

coiax committed:
Fixes misspelled define

duckay commented:
This is going to be fuckin hilarious when someone rolls para during combat

Bawhoppen commented:
@GuyonBroadway Part of ss13's fun is RNG. This is why disarms are still in the game.

GuyonBroadway commented:
@Bawhoppen

You can usually see a disarmer coming at you. It's the kind of RNG you can account for and doesn't leave you completly helpless.

This is a totally random litteraly out of nowhere permastun with no indication of what did it. It's less telegraphed that that wizard curse thats proposed by another PR.


People are going to ahelp over this because of how unfun and unintuitive it is.

coiax commented:
There is a 30% chance of picking a severe trauma.

There are 11 severe traumas (mutism, aphasia, blindness, paralysis, narcolepsy, monophobia, discoordination, pacifism, hypnotic stupor, split personality, split personality with keyword activation).

When paralysis is picked, the options are `pick("full","left","right","arms","legs","r_arm","l_arm","r_leg","l_leg")`, which is 9 options, only one of which is full body paralysis.

So, ignoring the chance of this event being selected in the first place, because calculating the likelihood of event selection accurately is semi-complicated (given it variesn since not all events are in the pool every time an event is selected, as well as a random time between events), we have:

3/10 * 1/11 * 1/9 = 1/330 chance of full body paralysis, which is 0.30%.

I don't think it's a big deal.

XDTM commented:
i mean, we kinda do have that

GuyonBroadway commented:
Wait really? That sounds fucking terrible.

ninjanomnom commented:
I have to agree that even a tiny chance for suddenly being unable to play the game isn't a good thing to have. I like the concept of this but some of the traumas need to be excluded.

coiax commented:
@XDTM Full body paralysis doesn't stop you talking or sending PDA messages, does it?

improvedname commented:
Cnn breaking news:murderbone decided to become a pacifist

SouDescolado commented:
Could it be made in such a way that if you have the suit sensors on it triggers an alarm for medical? Like "Its-a-Lizard needs medical evaluation" or something, so if someone is messing around in the maint and gets stuck somehow at least there is a chance someone will rescue(If the tracking is also on. Otherwise just alert).

tralezab commented:
Since the effects set in instantly you could make it skip people not on the station Z, miners getting the lottery die for no reason instantly while hilarious i'll concede probably should not happen

Bawhoppen commented:
I mean, this is too funny not to have in my opinion.
It's such a small chance, I don't see it as a big concern for balance. 
After all: "Losing is part of the game" and "Your character will frequently die, sometimes without even a possibility of avoiding it. Events will often be out of your control. No matter how good or prepared you are, sometimes you just lose."

ninjanomnom commented:
@coiax I can't get behind the chance for randomly being unable to play the game but triggering a brain degradation event which gives a trauma after a short period like what @XDTM said sounds like a good alternative.

GuyonBroadway commented:
@Bawhoppen rng death outta the blue isn't funny.

Losing is part of the game but usually each time you die you get an understanding of why it happened and learn to be aware of it for future. You can't account for everything but nothing hits you without any warning.

If you really think a random roll of the dice can just outright end someone then and there why not just cut out the middleman and have an event that literally just kills you.

Bawhoppen commented:
@GuyonBroadway I completely understand your stance, and I think it's an absolutely valid one. 
I am, however, approaching this from a different philosophy. One where I think your character shouldn't automatically be entitled to survive, even when you've done nothing wrong. 
ss13 is a game that can't (necessarily) be approached from traditional game design, such as where your level of skill affects your chances of survival. I see ss13 as 20-80 or so independent and personal stories of the crew aboard a space station, and how their stories all coalesce into one grander, singular narrative.  
So if it means that one person has to die, through no fault of their own, to create a better narrative for everyone else, I am okay with that. I believe the enjoyment of the overall playerbase takes precedence over a single individual's experience. 

Of course, we also don't want to go overboard with it, and start removing people from the game or fucking them over unnecessarily with no good reason to. But, I believe *sometimes* it's okay to do that. (BTW, we already have immovable rods that can randomly smack you in the face and gib you.)

Despite all that though, the more I've thought about this specific change, I don't really think that it's likely to contribute to a better overall story. 
I was initially imagining some murderboner who keels over in the heat of a fight, and how funny that would be to the ghosts orbitting them, or something other similar situation. But now, I just see it as more likely that some random chef will just fall to the floor while making burgers, and some other random person will end up dragging them to medbay. Not as interesting as I would've first hoped.

optimumtact commented:
@coiax this runtimes because the list you're giving to pickweight has numeric keys and it's making byond spazz 


[2019-01-01 10:49:51.281] runtime error: list index out of bounds
[2019-01-01 10:49:51.281] proc name: pickweight (/proc/pickweight)
[2019-01-01 10:49:51.281]   source file: _lists.dm,230
[2019-01-01 10:49:51.281]   usr: (src)
[2019-01-01 10:49:51.281]   src: null
[2019-01-01 10:49:51.281]   call stack:
[2019-01-01 10:49:51.281] pickweight(/list (/list))
[2019-01-01 10:49:51.281] /datum/round_event/brain_traum... (/datum/round_event/brain_trauma): traumatize(Johnny Gold (/mob/living/carbon/human))
[2019-01-01 10:49:51.281] /datum/round_event/brain_traum... (/datum/round_event/brain_trauma): start()
[2019-01-01 10:49:51.281] /datum/round_event/brain_traum... (/datum/round_event/brain_trauma): process()
[2019-01-01 10:49:51.281] Events (/datum/controller/subsystem/events): fire(0)
[2019-01-01 10:49:51.281] Events (/datum/controller/subsystem/events): ignite(0)
[2019-01-01 10:49:51.281] Master (/datum/controller/master): RunQueue()
[2019-01-01 10:49:51.281] Master (/datum/controller/master): Loop()
[2019-01-01 10:49:51.281] Master (/datum/controller/master): StartProcessing(0)
